core/object/script_language.h

```diff
@@ -221,20 +221,75 @@ struct ScriptCodeCompletionOption {
 		KIND_FILE_PATH,
 		KIND_PLAIN_TEXT,
 	};
+
+	enum Location {
+		LOCATION_LOCAL = 0,
+		LOCATION_PARENT_MASK = (1 << 8),
+		LOCATION_OTHER_USER_CODE = (1 << 9),
+		LOCATION_OTHER = (1 << 10),
+	};
+
 	Kind kind = KIND_PLAIN_TEXT;
 	String display;
 	String insert_text;
 	Color font_color;
 	RES icon;
 	Variant default_value;
 	Vector<Pair<int, int>> matches;
+	int location = LOCATION_OTHER;
 
 	ScriptCodeCompletionOption() {}
 
-	ScriptCodeCompletionOption(const String &p_text, Kind p_kind) {
+	ScriptCodeCompletionOption(const String &p_text, Kind p_kind, int p_location = LOCATION_OTHER) {
 		display = p_text;
 		insert_text = p_text;
 		kind = p_kind;
+		location = p_location;
+	}
+};
+
+const int KIND_COUNT = 10;
```

Comment:
If this is only used by the editor, it should not be here.

Reply:
Indeed, can be easily moved.